pkg/api/v1/clients/kube/crd/registry.go

```diff
@@ -0,0 +1,147 @@
+package crd
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/kubeutils"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiexts "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Registry struct {
```

Comment:
do these types and funcs need to be public? i don't think we should be accessing this registry anywhere outside of the CRD package, should we?